win: Fix reading from stdin

b-r-o-c-k opened the pull request.

justinmk commented:
@b-r-o-c-k ZyX might be busy this weekend, could you create a PR at Neovim which includes his PR? Then we can run CI on it.

justinmk commented on src/nvim/os/fileio.c:
```diff
@@ -284,14 +284,17 @@ ptrdiff_t file_read(FileDescriptor *const fp, char *const ret_buf,
-    if (fp->eof) {
+    if (fp->eof
+        // Allow only at most one os_read[v] call.
+        || called_read && fp->non_blocking) {
```
we usually prefer to be explicit about operator precedence:

    || (called_read && fp->non_blocking)) {

b-r-o-c-k committed:
win: Fix reading from stdin

* Reading from stdin on Windows is fixed in the same way as it was in
  #8267.
* The file_read function was returning without filling the
  destination buffer when it was called with a non-blocking file
  descriptor.
